Cesium.Preprocessor/MacroExpansionEngine.cs

```diff
@@ -268,6 +272,18 @@ IEnumerable<IToken<CPreprocessorTokenType>> ClearSpaceBuffer()
         }
         yield break;
 
+        void PoofComma()
+        {
+            foreach (var tok in lexer.EnumerateInReverse()) // look at previous tokens
+                if (tok.Text == ",") // is that a comma?
+                {
+                    MemoryMarshal.GetReference(tok.Text.AsSpan()) = ' '; // poof and there`s no more comma
```

Comment:
This modifies a `string` in-place which may have consequences. Please avoid doing that. All the compiler structures, including AST, lexer data and whatnot are immutable, not even starting about `string`.

Reply:
I don't see any other way to remove the previous token if we create a new collection via yield return that we don't have access to.

Reply:
Alright, I'll add this to my queue to rearchitecture the corresponding part.